tests/operator/helper.go

```diff
@@ -83,3 +88,112 @@ func OperatorInstalledMoreThanOnce(operator1, operator2 *provider.Operator) bool
+	for i := range allPods {
+		pod := allPods[i]
+		if pod.Namespace == namespace {
+			podsInNamespace = append(podsInNamespace, pod)
+		}
+	}
+	return podsInNamespace
+}
+
+func getCsvsBy(namespace string, allCsvs []*olmv1Alpha.ClusterServiceVersion) (csvsInNamespace []*olmv1Alpha.ClusterServiceVersion) {
+	for _, csv := range allCsvs {
+		if csv.Namespace == namespace {
+			csvsInNamespace = append(csvsInNamespace, csv)
+		}
+	}
+	return csvsInNamespace
+}
+
+func isSingleNamespacedOperator(operatorNamespace string, targetNamespaces []string) bool {
+	return len(targetNamespaces) == 1 && operatorNamespace != targetNamespaces[0]
+}
+
+func isMultiNamespacedOperator(operatorNamespace string, targetNamespaces []string) bool {
+	return len(targetNamespaces) > 1 && !stringhelper.StringInSlice(targetNamespaces, operatorNamespace, false)
+}
+
+func checkIfCsvUnderTest(csv *olmv1Alpha.ClusterServiceVersion) bool {
+	for _, testOperator := range env.Operators {
+		if testOperator.Csv.Name == csv.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func checkValidOperatorInstallation(namespace string) (isDedicatedOperatorNamespace bool, singleOrMultiNamespaceOperators,
+	nonSingleOrMultiNamespaceOperators, csvsFoundButNotInOperatorInstallationNamespace, operatorsFoundButNotUnderTest, podsNotBelongingToOperators []string) {
+	// 1. operator installation checks
+	csvsInNamespace := getCsvsBy(namespace, env.AllCsvs)
+
+	for _, csv := range csvsInNamespace {
+		// non empty when installed in the namespace
+		operatorNamespace := csv.Annotations["olm.operatorNamespace"]
+		targetNamespacesStr := csv.Annotations["olm.targetNamespaces"]
+
+		var targetNameSpaces []string
+		if len(targetNamespacesStr) != 0 {
+			targetNameSpaces = strings.Split(targetNamespacesStr, ",")
+		}
+
+		if namespace == operatorNamespace {
+			if checkIfCsvUnderTest(csv) {
+
+				isValidInstallation := isSingleNamespacedOperator(operatorNamespace, targetNameSpaces) || isMultiNamespacedOperator(operatorNamespace, targetNameSpaces)
+				if isValidInstallation {
+					singleOrMultiNamespaceOperators = append(singleOrMultiNamespaceOperators, csv.Name)
+				} else {
+					nonSingleOrMultiNamespaceOperators = append(nonSingleOrMultiNamespaceOperators, csv.Name)
+				}
+			} else {
+				operatorsFoundButNotUnderTest = append(operatorsFoundButNotUnderTest, csv.Name)
+			}
+		} else {
+			if len(targetNameSpaces) != 0 { // check for non-cluster wide operators
+				csvsFoundButNotInOperatorInstallationNamespace = append(csvsFoundButNotInOperatorInstallationNamespace, csv.Name)
+			}
+		}
+	}
+
+	// 2. existing pods check
+	podsNotBelongingToOperators, err := findPodsNotBelongingToOperators(namespace)
+	if err != nil {
+		return false, singleOrMultiNamespaceOperators, nonSingleOrMultiNamespaceOperators, csvsFoundButNotInOperatorInstallationNamespace, operatorsFoundButNotUnderTest, podsNotBelongingToOperators
```

Comment:
I would add another return (err) to report an internal error reason and display the err